### PR TITLE
feat(components): add custom xpath props support

### PR DESCRIPTION
**Pull request by akaseka**
**What is the problem this PR is trying to solve?**
Add custom xpath props support for listView, MultiselectTag, Bage components
https://jira.talendforge.org/browse/TDOPS-659

**What is the chosen solution to this problem?**
Add a new property -  dataTest

**Please check if the PR fulfills these requirements**

* [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/ListView/Items/Item/Item.component.js [hunk 1]
```diff
@@ -79,9 +79,10 @@ class Item extends Component {
 						</>
 					}
 					aria-label={ariaLabel}
+					data-test={`${dataTest}.checkbox`}
 				/>
 				{children && (
-					<div className={classNames('checkbox-nested-expand', { expanded: item.expanded })}>
+					<div className={classNames('checkbox-nested-expand', { expanded: item.expanded })} data-test={`${dataTest}.checkbox-nested-expand`}>
```
Comment (jsomsanith-tlnd) [inline packages/components/src/ListView/Items/Item/Item.component.js:85]: Can you check if `dataTest` is defined please ?

File: packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js [hunk 2]
```diff
@@ -244,7 +244,7 @@ export default class MultiSelectTag extends React.Component {
 				<div className={`${theme.wrapper} form-control`}>
 					{this.props.value.map((val, index) => {
 						const label = getLabel(this.getTitleMap(), val, names[index]);
-						const badgeProps = { label, key: index };
+						const badgeProps = { label, key: index, dataTest: `${this.schema.dataTest}.${index}` };
```
Comment (jsomsanith-tlnd) [inline packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js:247]: Same here please

File: packages/components/src/ListView/Items/Item/Item.component.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ class Item extends Component {
 	}
 
 	render() {
-		const { id, item, parentItem, isSwitchBox, searchCriteria, children, t } = this.props;
+		const { id, item, parentItem, isSwitchBox, searchCriteria, children, dataTest, t } = this.props;
```
Comment (inna-i) [inline packages/components/src/ListView/Items/Item/Item.component.js:24]: Please add a new prop into the Item.propTypes

File: packages/components/src/ListView/Items/Item/item.snapshot.test.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ describe('Item', () => {
 		const props = {
 			id: '0-item',
 			item: selectedItem,
+			dataTest:'item',
```
Comment (inna-i) [inline packages/components/src/ListView/Items/Item/item.snapshot.test.js:16]:
```suggestion
			dataTest: 'item',
```

**Review by jsomsanith-tlnd — APPROVED** [review]

**Review by inna-i — APPROVED** [review]

**Pull request merged**